patches/minecraft/net/minecraft/block/Block.java.patch

```diff
@@ -1354,6 +1354,24 @@
 +        return func_185467_w();
 +    }
 +
++    /**
++     * Gets the color this block adds to a beacon beam
++     *
++     * Return a float array with a red, green and blue element or null for no change
++     *
++     * The blending works like so: (old + new) / 2
```

Comment:
"The values returned are averaged with the RGB values the beam had before entering this block"

would be clearer in my opinion